sparta/sparta/resources/Buffer.hpp

```diff
@@ -191,14 +191,19 @@ namespace sparta
                 buffer_entry_(iter.buffer_entry_)
             {}
 
+            /**
+             * \brief a copy constructor that allows for implicit conversion from a
+             * regular iterator to a const_iterator.
```

Comment:
Yeah, good point.  It actually satisfies both requirements: 
1. Conversion: `BufferIterator<false> -> BufferIterator<true>`
2. Copy:  `BufferIterator<true> -> BufferIterator<true>`

I'll fix the comment.   This was added due to clang's clamping down on abuse of the rule of 3.  This one is tricky since we need both conversion and copy.